File: frontend/observable.ts
export class Observable {
    private observadores: Array<{ atualizar: (mensagem: string) => void }> = [];

    adicionarObservador(observador: { atualizar: (mensagem: string) => void }) {
        this.observadores.push(observador);
    }

    notificarObservadores(mensagem: string) {
        for (const observador of this.observadores) {
            observador.atualizar(mensagem);
        }
    }
}

File: src/biblioteca/Observable.java
package biblioteca;

import java.util.ArrayList;
import java.util.List;

public class Observable {
    private List<Observador> observadores = new ArrayList<>();

    public void adicionarObservador(Observador observador) {
        observadores.add(observador);
    }

    public void removerObservador(Observador observador){

    }

    public void notificarObservadores(Livro livro, Acervo acervo) {
        for (Observador observador : observadores) {
            observador.atualizar(livro, acervo);
        }
    }

    public Observador getPrimeirodaFila(){
        return this.observadores.get(0);
    }
}

interface Observador {
    void atualizar(Livro livro, Acervo acervo);
}
